main.py

```diff
@@ -483,30 +491,31 @@ def main(cl_arguments):
                 phase="target_train",
             )
 
-            # Now that we've trained a model, revert to the normal checkpoint
-            # logic for this task.
+            # Now that we've trained the task specific module for the task,
+            # since we are accumulating the best parameters for
+            # each task specific model, we allow for loading of the trained
+            # module. We avoid loading the task specific modules at first
+            # in order to make sure that any module-specific training from pretraining
+            # step does not affect the target task training step.
+            # This only affects for transfer_paradigm = frozen.
             if task.name in task_names_to_avoid_loading:
                 task_names_to_avoid_loading.remove(task.name)
 
-            # The best checkpoint will accumulate the best parameters for each
-            # task.
-            layer_path = os.path.join(args.run_dir, "model_state_target_train_best.th")
-
             if args.transfer_paradigm == "finetune":
-                # Save this fine-tune model with a task specific name.
-                finetune_path = os.path.join(args.run_dir, "model_state_%s_best.th" % task.name)
-                os.rename(layer_path, finetune_path)
-
                 # Reload the original best model from before target-task
-                # training.
-                pre_finetune_path = get_best_checkpoint_path(args.run_dir)
+                # training since we specifically finetune for each task.
+                pre_target_train = get_best_checkpoint_path(args.run_dir, "pretrain")
+                if len(pre_target_train) > 0:
```

Comment:
might as well move this check into `get_best_ckpt_path` if we're going to / want to check it every time?

Reply:
Realized it's already a part of the refactored function so deleting this redundant check.